### PR TITLE
Fixes #3944: Adding mass unit in "Base units"

### LINKED ISSUE #3944: mass unit not listed in 'Base units' table
It seems that the unit for the mass in MDAnalysis is missing. Should be unified atomic mass unit.

```rst
=========== ============== ===============================================
quantity    unit            SI units
=========== ============== ===============================================
mass         u              :math:`1.660538921 \times 10^{-27}` kg
```

### PR DESCRIPTION
**Pull request by 1e9abhi1e10**
Fixes #3944 

Changes made in this Pull Request:
 - Adding mass unit in "Base units"

PR Checklist
------------
 - [ ] Tests?
 - [ ] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

### CODE REVIEW

File: package/MDAnalysis/units.py
Comment (orbeckst) [inline package/MDAnalysis/units.py:331]: massUnit_factor is not defined anywhere

**Review by orbeckst — DISMISSED** [review]
Thank you for your PR. Sorry that nobody had a look at it in a while. I only could do a brief review. The main thing seems to be that it is currently not possible to do anything with the defined mass unit. There should be some test that make use of it.

I am asking @PicoCentauri as the original author of issue #3944 to also review as he should have a clearer idea of what should be present.

File: package/MDAnalysis/units.py
Comment (PicoCentauri) [inline package/MDAnalysis/units.py:46]: This where is belongs to. Thanks!

 However, I am not sure about the exact value of u. There was had a [comment](https://github.com/MDAnalysis/mdanalysis/issues/3944#issuecomment-1351805481) about a more accurate value. @v-parmar can you give a reference where we are using your value?

File: package/MDAnalysis/units.py
Comment (PicoCentauri) [reply package/MDAnalysis/units.py:331]: Yes, there could should be dictionary similar to the other. This would also define how we define our mass. Units could be `kg`, `g` and `u`


```python
massUnit_factor = {
    'u': 1.0,
    'kg': XXX
    'g': XXX,
}
```

**Comment by orbeckst** [comment]
I dismissed my own review. @PicoCentauri please decide how to proceed with this PR. Please either do a "Request Changes" with actionable questions or todos or if you're happy submit an "Approval" and merge. I think at the moment it is not clear to @1e9abhi1e10 what they need to do here.  Thanks!

File: package/MDAnalysis/units.py
Comment (PicoCentauri) [inline package/MDAnalysis/units.py:46]: We should the use the latest value of the mass here

```suggestion
   mass         u              :math:`1.66053906660 \times 10^{-27}` kg
```

File: package/MDAnalysis/units.py
Comment (PicoCentauri) [inline package/MDAnalysis/units.py:331]: I think this conversion is useful but you should add a dictionary as given below.

```python
massUnit_factor = {
    'u': 1.0,
    'kg': 1.6605390666e-27
    'g': 1.6605390666e-24,
}
```

**Review by PicoCentauri — CHANGES_REQUESTED** [review]
Thanks again @1e9abhi1e10 for this useful PR. In order to get this merged we have do to some minor changes. If you do not know how to implement them just let me know.

File: package/MDAnalysis/units.py
Comment (hmacdope) [reply package/MDAnalysis/units.py:46]: This is important change but tbh should be reflected in all units otherwise our unit system is inconsistent, see discussions about 2019 changes in #3941. Perhaps this is ok as a temporary fix?? What does @MDAnalysis/coredevs think?

Comment (hmacdope) [reply package/MDAnalysis/units.py:46]: I am going to block on this issue.  so people have a day or two to discuss.

**Review by hmacdope — CHANGES_REQUESTED** [review]
Block on unit consistency.

**Comment by orbeckst** [comment]
Are we continuing on this PR or should we close it? I am not sure what actually needs doing.

**Pull request closed without merging**
